rotkehlchen/chain/ethereum/modules/thegraph/decoder.py

```diff
@@ -0,0 +1,184 @@
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=ethereum_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.token = A_GRT.resolve_to_evm_token()
+
+    def counterparties(self) -> list[CounterpartyDetails]:
+        return [THEGRAPH_CPT_DETAILS]
+
+    def possible_events(self) -> DecoderEventMappingType:
+        return {CPT_THEGRAPH: {
+            HistoryEventType.STAKING: {
+                HistoryEventSubType.DEPOSIT_ASSET: EventCategory.DEPOSIT,
+                HistoryEventSubType.REMOVE_ASSET: EventCategory.WITHDRAW,
+            },
+            HistoryEventType.INFORMATIONAL: {
+                HistoryEventSubType.NONE: EventCategory.INFORMATIONAL,
+            },
+        }}
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            CONTRACT_THEGRAPH_STAKING: (self._decode_delegator_staking,),
+        }
+
+    def _decode_delegator_staking(self, context: DecoderContext) -> DecodingOutput:
+        """Processes events from The Graph's staking contract"""
+        if context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED:
+            return self._decode_stake_delegated(context)
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_LOCKED:
+            return self._decode_stake_locked(context)
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_WITHDRAWN:
+            return self._decode_stake_withdrawn(context)
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_stake_delegated(self, context: DecoderContext) -> DecodingOutput:
+        """Processes StakeDelegated event"""
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        stake_amount = hex_or_bytes_to_int(context.tx_log.data[:32])
+        stake_amount_norm = token_normalized_value(
+            token_amount=stake_amount,
+            token=self.token,
+        )
+        # identify and override the original stake Transfer event
+        for event in context.decoded_events:
+            if (
+                event.location_label == delegator and
+                event.address == CONTRACT_THEGRAPH_STAKING and
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                initial_amount_norm = event.balance.amount
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.balance = Balance(amount=stake_amount_norm)
+                event.notes = f'Delegated {stake_amount_norm} GRT to indexer {indexer}'
```

Comment:
Did not notice it before, but if you check most of our decoders we use present tense and start with a verb.

```suggestion
                event.notes = f'Delegate {stake_amount_norm} GRT to indexer {indexer}'
```